src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -72,15 +72,8 @@ seastar::future<> ClientRequest::start()
 	if (pg.can_discard_op(*m)) {
 	  return osd.send_incremental_map(conn.get(), m->get_map_epoch());
 	}
-	return with_blocking_future(
-	  handle.enter(pp(pg).await_map)
-	).then([this, &pg]() mutable {
-	  return with_blocking_future(
-	    pg.osdmap_gate.wait_for_map(m->get_map_epoch()));
-	}).then([this, &pg](auto map) mutable {
-	  return with_blocking_future(
-	    handle.enter(pp(pg).wait_for_active));
-	}).then([this, &pg]() mutable {
+	return with_blocking_future(handle.enter(pp(pg).wait_for_active))
```

Comment:
Who is responsible for ensuring an `OSDMap` in the required version has been dispatched to a `PG`? The `min_get_epoch()` works at the level of `osd.osdmap_gate`, not `pg.osdmap_gate`.

Reply:
how about 

1. removing the first phase waiting osd for getting the osdmap and 
1. instead adding a phase right before `wait_for_active` waiting for `pg.osdmap_gate.wait_for_map(m->get_min_epoch()`

Reply:
> 1. removing the first phase waiting osd for getting the osdmap and
> 2. instead adding a phase right before `wait_for_active` waiting for `pg.osdmap_gate.wait_for_map(m->get_min_epoch()`

I think this should work:-)

Reply:
> how about
> 
>     1. removing the first phase waiting osd for getting the osdmap and instead
> 
>     2. adding a phase right before `wait_for_active` waiting for `pg.osdmap_gate.wait_for_map(m->get_min_epoch()`

FWIW I don't see a reason why this shouldn't work.